File: 프로그래머스/Lv.0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

string solution(string my_string, vector<vector<int>> queries) {
    string answer = "";
    for(int i=0;i<queries.size();i++){
        string front = my_string.substr(0, queries[i][0]);
        string mid = my_string.substr(queries[i][0], queries[i][1] - queries[i][0] + 1);
        string back = my_string.substr(queries[i][1]+1, my_string.size() - queries[i][1]);
        reverse(mid.begin(),mid.end());
        my_string = front + mid + back;
    }
    return my_string;
}

File: 프로그래머스/Lv.0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.py
def solution(my_string, queries):
    answer = ''
    for q1, q2 in queries:
        mid = my_string[q1:q2+1]
        mid = mid[::-1]
        my_string = my_string[:q1] + mid + my_string[q2+1:]
    return my_string